vmdb/app/models/miq_worker.rb

```diff
@@ -91,6 +91,10 @@ def self.server_scope(server_id = nil)
     where(:miq_server_id => server_id)
   end
 
+  def self.no_subclasses_scope
+    where(:type => self.to_s)
+  end
+
   CONDITION_CURRENT = {:status => STATUSES_CURRENT}
   def self.find_current(server_id = nil)
     self.server_scope(server_id).where(CONDITION_CURRENT)
```

Comment:
@blomquisg doesn't `find_current` have the same problem?  do you want to tackle it here?  I think it makes sense.